### ISSUE TITLE
[HxInputTags] - it is impossible to select another tag when input is focused

### ISSUE THREAD

**Issue by pavmal12** [havit/Havit.Blazor#394]
After I selected one tag from dropdown, I am not able to open dropdown to select another tag while input keeps focused. 

Repro steps:
1. Click on input
2. select any tag from dropdown
3. click again on input 

Actual result: the dropdown with tags will not open

Expected result: I expect the dropdown to be opened

Workaround: 
I need click outside of the input and then I will be able to open dropdown again. 

![Screen Recording 2022-09-26 at 15 46 26](https://user-images.githubusercontent.com/110027149/192293444-ec0d3f02-f804-445d-9333-52fed4c3fd19.gif)

**Comment by tommi1hirvonen** [comment]
I'm using `HxInputTags` quite heavily in my application as well, and if I'm not too mistaken, I think this might have worked differently in a previous version. For some reason I remember at one time the input working in such a way that after having selected one tag from the dropdown menu, the menu would remain open (or rather reopen) automatically. Currently it is behaving as described by @pavmal12. I too preferred the old way which allowed easy selection of multiple tags.

**Comment by tommi1hirvonen** [comment]
I did some investigating and actually found the one line of code that makes it work.
```
private async Task<InputTagsDataProviderResult> GetNewItemSuggestions(InputTagsDataProviderRequest request)
{
    await Task.Delay(50); // <= WHEN THIS LINE IS REMOVED, THE DROPDOWN MENU IS NOT REOPENED.
    return new InputTagsDataProviderResult()
    {
	Data = Enum.GetValues<ThemeColor>().Select(v => v.ToString()).Where(v => v.Contains(request.UserInput, StringComparison.OrdinalIgnoreCase))
    };
}
```
So if the method is not actually awaiting anything, then the tag dropdown menu is not reopened after selecting one tag. If you wait for a delay of say 50 ms, then the dropdown menu is reopened.

I've noticed similar behaviour with the `HxAutosuggest` component, where not having any long running operations in the data provider method sometimes makes the dropdown disappear after the input receives focus. As a workaround I've placed a small delay in the data provider methods (e.g. 150 ms), which makes the autosuggest dropdown behave more consistently.

Optimally the dropdowns would work in the same way even without the delays, as in my case there's no long running operation to get either the tag or autosuggest suggestions. Now I'm forced to add an artificial delay to the method to make the dropdowns behave nicer.
